tests/deepsparse/pipelines/test_pipeline.py

```diff
@@ -141,3 +142,21 @@ def sleep_for_10ms(xs):
         end = time.time_ns()
         dur_ms = (end - start) * 1e-6
         assert abs(dur_ms - 30) < 10
+
+
+@mock_engine(rng_seed=0)
+def test_run_with_monitoring(engine_mock):
```

Comment:
nit: probably would want to make each of the pre/post/forward functions take a specific amount of time known by the unit test to truly verify that our timings are correct.